CHANGELOG.md

```diff
@@ -118,6 +118,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Add support for aggregation profiler with concurrent aggregation ([#8801](https://github.com/opensearch-project/OpenSearch/pull/8801))
 - [Remove] Deprecated Fractional ByteSizeValue support #9005 ([#9005](https://github.com/opensearch-project/OpenSearch/pull/9005))
 - Make MultiBucketConsumerService thread safe to use across slices during search ([#9047](https://github.com/opensearch-project/OpenSearch/pull/9047))
+- Rethrow OpenSearch exception in concurrent search ([#9177](https://github.com/opensearch-project/OpenSearch/pull/9177))
```

Comment:
I think we can make this changelog description more precise -- the issue here is that we wrap `OpenSearchException` in the non-concurrent path with concurrent search.